PurchasesTests/Networking/BackendTests.swift

```diff
@@ -475,28 +475,28 @@ class BackendTests: XCTestCase {
     }
 
     func testPayAsYouGoPostsCorrectly() throws {
-        let response = HTTPResponse(statusCode: 200, response: validSubscriberResponse, error: nil)
+        let response = HTTPResponse(statusCode: .success, response: validSubscriberResponse, error: nil)
         httpClient.mock(requestPath: .postReceiptData, response: response)
         postPaymentMode(paymentMode: .payAsYouGo)
         try checkCall(expectedValue: 0)
     }
 
     func testPayUpFrontPostsCorrectly() throws {
-        let response = HTTPResponse(statusCode: 200, response: validSubscriberResponse, error: nil)
+        let response = HTTPResponse(statusCode: .success, response: validSubscriberResponse, error: nil)
         httpClient.mock(requestPath: .postReceiptData, response: response)
         postPaymentMode(paymentMode: .payUpFront)
         try checkCall(expectedValue: 1)
     }
 
     func testFreeTrialPostsCorrectly() throws {
-        let response = HTTPResponse(statusCode: 200, response: validSubscriberResponse, error: nil)
+        let response = HTTPResponse(statusCode: .success, response: validSubscriberResponse, error: nil)
         httpClient.mock(requestPath: .postReceiptData, response: response)
         postPaymentMode(paymentMode: .freeTrial)
         try checkCall(expectedValue: 2)
     }
 
     func testForwards500ErrorsCorrectlyForCustomerInfoCalls() {
-        let response = HTTPResponse(statusCode: 501, response: serverErrorResponse, error: nil)
+        let response = HTTPResponse(statusCode: .internalServerError, response: serverErrorResponse, error: nil)
```

Comment:
I wonder why we had 501, it was probably an accident since I don't think we ever expect to get `Not Implemented`

Reply:
I left these to make sure that implementations check more than 500 but the entire range.